src/source/oscap_source.c

```diff
@@ -358,8 +358,16 @@ int oscap_source_validate(struct oscap_source *source, xml_reporter reporter, vo
 
 int oscap_source_validate_schematron(struct oscap_source *source, const char *outfile)
 {
-	return oscap_source_validate_schematron_priv(source, oscap_source_get_scap_type(source),
-			oscap_source_get_schema_version(source), outfile);
+	FILE *outfile_fd = stdout;
+	if (outfile == NULL)
```

Comment:
This function is called only at 2 places and always with outfile = NULL. I don't see a benefit in this parameter. Do you think we can mark this function as deprecated in the API and introduce a new function? Or would it be a "scope creep"?

Reply:
I see it as a poor man's way to return the validation result back to the user's code without capturing the output. Coming up with a better return mechanism would definitely be a scope creep. 

On the other hand marking it as deprecated with a further note (maybe in a form of an issue) to provide a better way of doing the same should be OK, IMHO.

Reply:
But you shouldn't add something as OSCAP_DEPRECATED without providing a new solution at the same time because it would cause compiler warnings at the 2 places where the old function is called.

Reply:
I would vote for a different PR then.

Reply:
OK

Reply:
https://github.com/OpenSCAP/openscap/issues/1720